app/src/main/java/com/sun/dummyshop/ui/cart/CartViewModel.kt

```diff
@@ -0,0 +1,88 @@
+package com.sun.dummyshop.ui.cart
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import com.sun.dummyshop.base.BaseViewModel
+import com.sun.dummyshop.data.model.Bill
+import com.sun.dummyshop.data.model.CartProduct
+import com.sun.dummyshop.data.model.CheckoutBody
+import com.sun.dummyshop.data.model.Product
+import com.sun.dummyshop.data.repository.BillRepository
+import com.sun.dummyshop.data.repository.ProductRepository
+import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers
+import io.reactivex.rxjava3.kotlin.addTo
+import io.reactivex.rxjava3.schedulers.Schedulers
+
+class CartViewModel(
+    private val productRepository: ProductRepository,
+    private val billRepository: BillRepository
+) : BaseViewModel() {
+
+    private val _products = MutableLiveData<List<Product>>()
+    val products: LiveData<List<Product>>
+        get() = _products
+
+    private val _bill = MutableLiveData<Bill>()
+    val bill: LiveData<Bill>
+        get() = _bill
+
+    private val _total = MutableLiveData<Int>()
+    val total: LiveData<Int>
+        get() = _total
+
+    private fun saveBill(bill: Bill) {
+        billRepository.insertBill(bill)
+            .subscribeOn(Schedulers.io())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe({
+            }, {
+                _error.value = it.message.toString()
+            })
+            .addTo(disposable)
+    }
+
+    private fun calculateAmount() {
+        var totalAmount = 0
+        products.value?.let { list ->
+            list.forEach {
+                totalAmount += it.price.toInt() * it.quantityInCart
+            }
+        }
+        _total.value = totalAmount
+    }
+
+    fun getProductsAddedToCart() {
+        productRepository.getAddedToCartProducts()
+            .subscribeOn(Schedulers.io())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe({
+                _products.value = it
+                calculateAmount()
+            }, {
+                _error.value = it.message.toString()
+            })
+            .addTo(disposable)
+    }
+
+    fun checkout() {
+        productRepository.getAddedToCartProducts()
+            .map {
+                it.map { product ->
+                    CartProduct(product.id, product.quantityInCart.toString())
+                }
+            }
+            .flatMap {
+                val checkoutBody = CheckoutBody(it)
+                billRepository.checkout(checkoutBody)
+            }
+            .subscribeOn(Schedulers.io())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe({
+                _bill.value = it
```

Comment:
```suggestion
                _bill.postValue(it)
```
trường hợp này thì nên dùng post nhé

Reply:
vâng